File: custom_components/fleet_charging/panel.py
import os
import logging
from homeassistant.components.http import HomeAssistantView
from homeassistant.core import HomeAssistant

_LOGGER = logging.getLogger(__name__)

class FleetChargingPanel(HomeAssistantView):
    """Vytvorenie API endpointu pre vizualizáciu v integrácii."""

    url = "/api/fleet_charging"
    name = "api:fleet_charging"
    requires_auth = True

    def __init__(self, hass: HomeAssistant):
        """Inicializácia s referenciou na Home Assistant."""
        self.hass = hass
        self.db = hass.data["fleet_charging"]["db"]

    async def get(self, request):
        """Načítanie všetkých údajov potrebných pre panel."""
        try:
            users = await self.db.get_all_users()
            vehicles = await self.db.get_all_vehicles()
            sessions = await self.db.get_all_sessions()

            return self.json({
                "users": users,
                "vehicles": vehicles,
                "sessions": sessions
            })
        except Exception as e:
            _LOGGER.error(f"Chyba pri načítaní údajov: {e}")
            return self.json({"error": "Nepodarilo sa načítať údaje"}, status=500)

async def async_setup_panel(hass: HomeAssistant):
    """Registrácia API panelu v Home Assistant."""
    hass.http.register_view(FleetChargingPanel(hass))

File: custom_components/fleet_charging/panel.js
class FleetChargingPanel extends HTMLElement {
    constructor() {
        super();
        this.attachShadow({ mode: "open" });
    }

    set hass(hass) {
        this.hass = hass;
        this.fetchData();
    }

    async fetchData() {
        try {
            const response = await fetch("/api/fleet_charging");
            this.data = await response.json();
            this.render();
        } catch (error) {
            console.error("Chyba pri načítaní údajov:", error);
        }
    }

    render() {
        if (!this.shadowRoot) return;

        this.shadowRoot.innerHTML = `
            <style>
                .container { padding: 20px; font-family: Arial, sans-serif; }
                h2 { color: #3498db; }
                .section { margin-bottom: 20px; padding: 10px; border: 1px solid #ddd; border-radius: 5px; background: #f9f9f9; }
                .section h3 { margin-top: 0; }
                .item { padding: 5px 0; }
            </style>
            <div class="container">
                <h2>Fleet Charging Manager</h2>
                <div class="section">
                    <h3>🔹 Priradenie používateľov</h3>
                    ${this.data?.users?.map(user => `<div class="item">👤 ${user.name}</div>`).join('') || "Žiadni používatelia"}
                </div>
                <div class="section">
                    <h3>🚗 Vozidlá</h3>
                    ${this.data?.vehicles?.map(vehicle => `<div class="item">🚘 ${vehicle.name}</div>`).join('') || "Žiadne vozidlá"}
                </div>
                <div class="section">
                    <h3>⚡ Nabíjacie relácie</h3>
                    ${this.data?.sessions?.map(session => `<div class="item">🔋 ${session.vehicle_id} nabíjané užívateľom ${session.user_id}</div>`).join('') || "Žiadne relácie"}
                </div>
            </div>
        `;
    }
}

customElements.define("fleet-charging-panel", FleetChargingPanel);
